test/ca-setup.sh

```diff
@@ -0,0 +1,41 @@
+#! /usr/bin/env bash
+set -e
+rm -f test/nginx/self.test.* test/ca.* ../certificates/ca.crt || true
+cp test/cert-test.py sentry/
+mkdir -p certificates/
+
+openssl req -x509 -new -nodes -newkey rsa:2048 -keyout test/ca.key \
+ -sha256 -days 1 -out test/ca.crt -batch  \
+-subj "/CN=TEST CA *DO NOT TRUST*" \
+-addext "keyUsage = critical, keyCertSign, cRLSign" \
+-addext "nameConstraints = critical, permitted;DNS:self.test"
+
+# openssl x509 -in test/ca.crt -text -noout
+cp test/ca.crt certificates/
+
+openssl req -new -nodes -newkey rsa:2048 -keyout test/nginx/self.test.key \
+-addext "subjectAltName=DNS:self.test" \
+-out test/nginx/self.test.req -batch -subj "/CN=Self Signed with CA Test Server"
+
+# openssl req -in test/nginx/self.test.req -text -noout
+
+openssl x509 -req -in test/nginx/self.test.req -CA test/ca.crt -CAkey test/ca.key \
+-extfile <(printf "subjectAltName=DNS:self.test") \
+-CAcreateserial -out test/nginx/self.test.crt -days 1 -sha256 
+
+# openssl x509 -in test/nginx/self.test.crt -text -noout
+openssl verify -CAfile test/ca.crt test/nginx/self.test.crt
+
+openssl req -x509 -newkey rsa:2048 -nodes -days 1 -keyout test/nginx/fake.test.key \
+-out test/nginx/fake.test.crt -addext "subjectAltName=DNS:fake.test" -subj "/CN=Self Signed Test Server"
+# openssl x509 -in test/nginx/fake.test.crt -text -noout
+
+# docker-compose up -d test
+# docker-compose restart test
+
+# # curl https://self.test --cacert ca.crt --resolve self.test:443:$(docker inspect sentry_onpremise_test_1| jq .[0].NetworkSettings.Networks.sentry_onpremise_default.IPAddress -r)
+# curl https://self.test  --cacert certificates/ca.crt --resolve self.test:443:$(docker inspect sentry_onpremise_test_1| jq .[0].NetworkSettings.Networks.sentry_onpremise_default.IPAddress -r)
+# cp ./test/cert-test.py ./sentry/
+# # docker-compose run --rm web python3 /etc/sentry/cert-test.py
+# docker-compose down
+# rm -f ca.* certificates/* test/nginx/*.test.* sentry/cert-test.py 
```

Comment:
Lots of commented-out code here. Could we remove those and add some meaningful comments instead? What exactly are we testing here? Or is this just setup for a `cert-test.py`?

Reply:
Left over test code mistakenly committed, I'll sprinkle in a few notes.